node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -135,6 +131,20 @@ impl ParticipationRequest {
 		let Self { candidate_hash, candidate_receipt, .. } = self;
 		(candidate_hash, candidate_receipt)
 	}
+	// For tests we want to check whether requests are equal, but the
+	// request_timer field of ParticipationRequest doesn't implement 
+	// eq. This helper checks whether all other fields are equal,
+	// which is sufficient.
+	#[cfg(test)]
+	pub fn functionally_equal(&self, other: ParticipationRequest) -> bool {
+		if  &self.candidate_receipt == other.candidate_receipt() &&
+			&self.candidate_hash == other.candidate_hash() &&
+			self.session == other.session()
```

Comment:
I would destructure `self` into its fields, so that if a field is ever added in the future we don't forget to add it to the comparison here. It would look something like this:

```rust
let ParticipationRequest { candidate_receipt, candidate_hash, session, _request_timer } = self;
```

Credit to @eskimor for teaching me this!

Reply:
nit: Also, instead of `if cond { return true; } false`, you can simplify it to just `cond`. :) I think clippy should warn about this.

Reply:
> Credit to @eskimor for teaching me this!

+1 :)